instagram/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import Loader from 'react-loader-spinner';
+import dummyData from '../../dummy-data';
+import PostContainer from '../PostContainer/PostContainer';
+import SearchBar from '../SearchBar/SearchBar';
+import AccountMenu from './AccountMenu/AccountMenu';
+
+export default class PostPage extends Component {
+    constructor() {
+        super();
+        this.state = {
+            liked: false,
+            searchValue: '',
+            account: false,
+            data: []
+        };
+    }
+
+    componentDidMount() {
+        this.setState({
+            data: localStorage.getItem('data') === null ? dummyData : JSON.parse(localStorage.getItem('data'))
+        });
+    }
+
+    submitComment = (timestamp, comment) => {
+        const commentMatch = this.state.data[this.state.data.map((obj, i) => obj.timestamp === timestamp && i).filter(arr => arr !== false).pop()];
+
+        commentMatch.comments.push(comment);
+
+        localStorage.setItem('data', JSON.stringify(this.state.data))
+    }
+
+    search = e => {
+        this.setState({ [e.target.name]: e.target.value });
+
+        const search = this.state.data.filter(post => post.username.toLowerCase().includes((e.target.value).toLowerCase()) || post.comments.map(comm => comm.text.toLowerCase().includes((e.target.value).toLowerCase()) || comm.username.toLowerCase().includes((e.target.value).toLowerCase())) === true);
+
+        search.length !== 0 && this.setState({ data: search });
+
+        if (e.target.value.length === 0) {
+            this.setState({ data: dummyData });
+        }
+    }
```

Comment:
Super job making the search functionality to work. You might want to look for edge cases for the logic here. What if a user want to search for `fortnite` but misspell it as `pfortnite` then using the cursor to the left, delete the letter `p`. I don't think the post from `fortnite` will show up in this case.
Note: this isn't something you really need to worry about. It is fine just the way it is, I'm just being super picky because I see you like to tinkering with the logic.